### DISCUSSION TITLE
I tried to implement localsend protocol v2 in rust

### DISCUSSION THREAD

**Discussion by tom8zds** [localsend/localsend#1076] [Discussion (English, International)]
It seams that dart has performance issue on receiving. I've tested dart to rust and dart to dart

dart to rust

![image](https://github.com/localsend/localsend/assets/20534594/7dbb54da-ed11-4313-96a8-6aea4cd05754)

dart to dart 

![屏幕截图 2024-01-06 221754](https://github.com/localsend/localsend/assets/20534594/97caf4e2-c42c-490e-988e-0e0d8e744279)

**Comment by Tienisto** [comment]
Is it Windows to Android or Windows to another PC?
Can you try to disable encryption?

**Reply by tom8zds** [reply]
Andorid to windows, disable encryption does improves a bit, but still has disparity to rust implementation
![image](https://github.com/localsend/localsend/assets/20534594/2c1c82e6-755f-4537-a4f9-c12c374688d7)

**Reply by Tienisto** [reply]
Thank you. So if I understood correctly:
- Android to your Rust implementation achieves 500 Mbit
- Android to the current Dart implementation achieves 150 Mbit (or 200 Mbit if using http)

Is it correct?

**Reply by tom8zds** [reply]
I tested more rounds, network statistic seems has some bug, they can all acheive 90MB/s+ disk speed, which is 720Mbps network speed, but rust implementation can some times goes to 100MB/s+,  which is 800Mbps network speed. result is rust is about 10MB faster than dart
![image](https://github.com/localsend/localsend/assets/20534594/13e78a1c-feb9-4924-a5b9-c64625bffe68)
![image](https://github.com/localsend/localsend/assets/20534594/b360b91c-b1d2-4d32-b85e-a8d8baa2928d)
